### PR TITLE
fix pep8 and fix #1730 : improvements on the OSM downloader

### LINKED ISSUE #1730: OSM downloader layout and behaviour needs improvements
# Problem

The layout since adding the points in buildings no longer properly scales the extent widget to the dialog width.

inasafe openstreetmap downloader_172

The 'select on map' button should be placed in between the other widgets in order to be consistent with the extent selector dialog - see screenie below.

inasafe analysis area_174

Downloading for a small area (with extents as depicted in the first screenshot above) fails with the error message below:

selection_173

# Proposed solution

* Ensure extent selector part of the widget is in a grid layout
* Move the button down in between the others
* Investigate the download error and see if you can replicate it.

### PR DESCRIPTION
**Pull request by Gustry**
fix pep8 and fix #1730 : improvements on the OSM downloader : 
  - fix horizontal layout on the OSM downloader
  - fix horizontal layout on the extent selector dialog
  - fix error message when no OSM data
  - move the button down about "select on map"

### CODE REVIEW

**Comment by Gustry** [comment]
test this please, jenkins

File: safe/gui/tools/osm_downloader_dialog.py [hunk 1]
```diff
@@ -488,7 +492,8 @@ def load_shapefile(self, feature_type):
 
         if not os.path.exists(path):
             message = self.tr(
-                "%s don't exist. The server doesn't have any data.")
+                "%s doesn't exist. The server doesn't have any data on this "
```
Comment (timlinux) [inline safe/gui/tools/osm_downloader_dialog.py]: .... any data for this extent....

**Comment by timlinux** [comment]
LGTM when small grammar issue is resolved

**Comment by Gustry** [comment]
osm-downloader1

**Pull request merged**